src/dune_rules/module_compilation.ml

```diff
@@ -11,7 +11,7 @@ module CC = Compilation_context
 let force_read_cmi ~(cm_kind : Lib_mode.Cm_kind.t) source_file =
   let args = [ "-intf-suffix"; Path.extension source_file ] in
   match cm_kind with
-  | Melange Cmj -> "-bs-read-cmi" :: args
+  | Melange Cmj -> "--bs-read-cmi" :: args
```

Comment:
I guess this is an unrelated change. separate PR if possible?

Reply:
https://github.com/ocaml/dune/pull/6365